relay/sources/relayd/src/api/shared_folder.rs

```diff
@@ -1,13 +1,56 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 // SPDX-FileCopyrightText: 2019-2020 Normation SAS
 
-use crate::{error::Error, hashing::Hash, JobConfig};
-use futures::{future, Future};
+use crate::{api::RudderReject, hashing::Hash, JobConfig};
+use anyhow::Error;
 use serde::Deserialize;
 use std::{io, path::PathBuf, sync::Arc};
 use tokio::fs::read;
-use tracing::{debug, span, trace, Level};
-use warp::http::StatusCode;
+use tracing::{debug, error, span, trace, Level};
+use warp::{filters::method, fs, http::StatusCode, path, query, Filter, Reply};
+
+pub fn routes_1(
+    job_config: Arc<JobConfig>,
+) -> impl Filter<Extract = impl Reply, Error = warp::Rejection> + Clone {
+    let base = path!("shared-folder" / ..);
+
+    let job_config_head = job_config.clone();
+    let head = method::head()
+        .and(base)
+        .map(move || job_config_head.clone())
+        .and(path::peek())
+        .and(query::<SharedFolderParams>())
+        .and_then(|j, p, q| handlers::head(p, q, j));
+
+    let job_config_get = job_config;
```

Comment:
does this variable have a purpose?

Reply:
not this one indeed, as it is the last one and it is not moved by a closure.